src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -0,0 +1,140 @@
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Deletes a tag from an existing person in the address book, if the tag exists.
+ */
+public class DeleteTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "untag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes a tag from the tags of an existing contact, "
+            + "as specified by the index number used in the displayed person list. The tag will be deleted only if "
+            + "the tag is valid and is tagged to the contact (case-insensitive).\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TAG + "TAG]\n"
+            + "Example: " + COMMAND_WORD + " 2 "
+            + PREFIX_TAG + "friends";
+
+    public static final String MESSAGE_DELETE_TAG_SUCCESS = "Deleted tag: %1$s";
+    public static final String MESSAGE_MISSING_TAG = "This person is not tagged to %1$s.";
+
+    private final Index index;
+    private final Tag tag;
+
+    /**
+     * Constructs an {@code DeleteTagCommand} with the given {@code Index} and {@code Tag}.
+     *
+     * @param index of the person in the filtered person list to add the {@code Tag}.
+     * @param tag to be deleted from the {@code Person} specified by {@code index}.
+     */
+    public DeleteTagCommand(Index index, Tag tag) {
+        requireNonNull(index);
+        requireNonNull(tag);
+
+        this.index = index;
+        this.tag = tag;
+    }
+
+    /**
+     * Deletes a tag from an existing person in the address book, if the tag exists.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return the command result after the command execution.
+     * @throws CommandException if the {@code Tag} do not exist or the {@code Index} is invalid.
+     */
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        if (!personToEdit.hasTag(this.tag)) {
+            throw new CommandException(String.format(MESSAGE_MISSING_TAG, this.tag));
+        }
+
+        Person editedPerson = createPersonWithDeletedTag(personToEdit, this.tag);
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_DELETE_TAG_SUCCESS, editedPerson));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of personToEdit {@code Person}
```

Comment:
Maybe need to change this javadoc to follow the format? But aeems like it's passing the checkstyle